dev-docs/RFCs/v4.0.0/polyfill-modes-rfc.md

```diff
@@ -12,10 +12,118 @@ Our current polyfill algorithm allocates cells to polygons based on whether the
+and pentagons. The `polyfill` function is different that it assumes Cartesian geometry. For
+consistency with the rest of the library, the `polyfill` functions should be able to use the same
+cell boundaries.
+
+Maintaing a Cartesian option is useful for cases where polygons have been drawn on a projected map
+and the boundaries should be the same.
+
+* Very large polygons
+
+Polyfills of very large polygons require allocating large blocks of memory, and spending large
+amounts of time in a library function without progress information being available to the caller.
+(To be determined if this is in scope for this RFC or for another.)
+
 ## Approaches
 
 *What are the various options to address this issue?*
 
+On an API level, we will need to expose the containment mode and spherical/Cartesian choice as
+options to the caller.
+
+Internally, we would like to reuse the same implementation as much as possible, but change the
+exact inclusion check used for each mode.
+
+### Comparisons
+
+#### S2
+
+Contains separate functions for [intersection and containment](http://s2geometry.io/devguide/basic_types#s2polygon)
+
+#### Turf library
+
+Contains separate functions for [intersection](http://turfjs.org/docs/#booleanIntersects), [containment](http://turfjs.org/docs/#booleanContains), etc.
+
+#### PostGIS
+
+Contains separate functions for [intersection](https://postgis.net/docs/ST_Intersects.html), [containment](https://postgis.net/docs/ST_Contains.html), etc.
+
+See also [Simple Feature Access - SQL](https://www.ogc.org/standards/sfs).
+
+#### JTS
+
+Separate predicatess. (Reference: [JTS Developer Guide](https://github.com/locationtech/jts/blob/master/doc/JTS%20Developer%20Guide.pdf))
+
+#### GeoPandas
+
+Contains separate functions for [intersection](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.intersection.html), [containment](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.contains.html), etc.
+
 ## Proposal
 
-*What is the recommended approach?*
\ No newline at end of file
+*What is the recommended approach?*
+
+The signature for `polygonToCells` and `maxPolygonToCellsSize` would be changed as follows:
+
+```
+/** @brief maximum number of hexagons that could be in the geoloop */
+DECLSPEC H3Error H3_EXPORT(maxPolygonToCellsSize)(const GeoPolygon *geoPolygon,
+                                                  int res, uint32_t flags, int64_t *out);
+
+/** @brief hexagons within the given geopolygon */
+DECLSPEC H3Error H3_EXPORT(polygonToCells)(const GeoPolygon *geoPolygon,
+                                           int res, uint32_t flags, H3Index *out);
+```
+
+`flags` would have the following possible bit layout:
+
+|       Bits | Meaning
+| ---------- | -------
+| 1-2 (LSB)  | If 0, containment mode centroid.<br>If 1, containment mode cover.<br>If 2, containment mode intersects.<br>3 is a reserved value.
```

Comment:
As discussed previously, were we going to swap this around to put the cartesian/spherical at the LSB and allow this section more room to easily expand?

Also, are we planning to have boolean flags for each mode (i.e. each mode has one bit and we check LTR using simple AND checks) or are we setting aside a multi-bit section that we convert to an integer to check an enum value?

Reply:
I think I updated thison the RFC branch but this PR wasn't rebased on it (or I didn't push it?)